packages/xstate-inspect/test/inspect.test.ts

```diff
@@ -141,4 +141,46 @@ describe('@xstate/inspect', () => {
       }
     });
   });
+
+  it('should not crash when registering machine with very deep context when serializer manages to replace it', (done) => {
```

Comment:
I've analyzed this situation further and it turned out that this wasn't about circular references at all. The only required prerequisite for the crash was a very deep object in context.